lib/campaign.js

```diff
@@ -36,8 +36,14 @@ class Campaign {
   }
 
   getOne(id, appId, cb) {
-    if (!id || typeof id === 'function') {
-      return cb(new Error('id parameter must be provided.'))
+    if (typeof id === 'function') {
+      return id(
+        new Error('id parameter must be provided, as the first argument'),
+      )
+    }
+
+    if (!id) {
+      throw new Error('id parameter must be provided.')
```

Comment:
Should this be returning a failed promise? Should we always be returning a promise from these functions?

Reply:
Yeah, I think you're right. Since the function is not explicitly `async`, just throwing might create control flow errors.